test/pypi-integration.py

```diff
@@ -92,31 +93,25 @@ def test_fail_to_upload_if_input_name_different_from_package_name(self):
                 }
             )
 
-    def test_upload_when_different_names_allowed(self):
-        rc = subprocess.run(['python', 'setup.py', 'sdist'], check=True, cwd=REPODIR)
-        print("sdist returned", rc)
-        with self.assertRaises(out.NamesValidationError) as context:
-            out.out(
-                os.path.join(REPODIR, 'dist'),
-                {
-                    'source': {'test': True, 'name': 'mismatching-name', 'name_must_match': False},
-                    'params': {'glob': '*.tar.gz'}
-                }
-            )
-
     def test_fail_to_upload_if_package_version_not_pep440_compliant(self):
-        rc = subprocess.run(['python', 'setup.py', 'sdist'], check=True, cwd=os.path.join(THISDIR, 'test_package1_4'))
+        rc = subprocess.run(['python', 'setup.py', 'sdist'],
+            check=True, cwd=os.path.join(THISDIR, 'generalized_package'),
+            env={
+                **os.environ,
+                'TEST_PACKAGE_NAME': 'test_package1',
+                'TEST_PACKAGE_VERSION': '0.0.0-343-gea3bdad',
+            }
+        )
         print("sdist returned", rc)
         with self.assertRaises(out.VersionValidationError):
             out.out(
-                os.path.join(THISDIR, 'test_package1_4'),
```

Comment:
This generalized package (4 new files in test/generalized_package/) is introduced to:

to prevent that its not being built by make test and used by mistake in nexus-integration tests
to avoid duplication and allow easy override for the package name and version directly from the tests. See test_fail_to_upload_if_package_version_not_pep440_compliant. This will allow later (next PR) to add additional tests with the same package files.

It replaces: `test/test_package1_4`

Further improvements are possible here, I am considering another PR to improve the nexus and pypi integration tests.